tasks_test.go

```diff
@@ -127,71 +138,88 @@ func runInlineIngestionTask[T any](client *Client, dataSourceName string, entrie
 		WaitForService("middlemanager", wait.NewHTTPStrategy(processInformationPathPrefix).WithPort("8091/tcp").WithStartupTimeout(180*time.Second)).
 		Up(ctx, tc.Wait(true))
-	require.NoError(t, err, "druid services should be up with no error")
-
-	// Test create ingestion task -> get status -> complete sequence.
-	runInlineIngestionTask(d, "test-submit-task-datasource", testObjects, 2)
-	require.NoError(t, err, "error should be nil")
+	s.Require().NoError(err, "druid services should be up with no error")
 }
 
-func TestTerminateTask(t *testing.T) {
-	// Set up druid containers using docker-compose.
-	compose, err := tc.NewDockerCompose("testdata/docker-compose.yaml")
-	require.NoError(t, err, "NewDockerComposeAPI()")
+func (s *TasksServiceTestSuite) TearDownSuite() {
+	s.Require().NoError(
+		s.compose.Down(
+			context.Background(),
+			tc.RemoveOrphans(true),
+			tc.RemoveVolumes(true),
+			tc.RemoveImagesLocal),
+		"compose.Down()",
+	)
+}
 
-	// Set up cleanup for druid containers.
-	t.Cleanup(func() {
-		require.NoError(t, compose.Down(context.Background(), tc.RemoveOrphans(true), tc.RemoveVolumes(true), tc.RemoveImagesLocal), "compose.Down()")
-	})
+func (s *TasksServiceTestSuite) TestSubmit() {
+	// Test create ingestion task -> get status -> complete sequence.
+	err := runInlineIngestionTask(s.client, "test-submit-task-datasource", testObjects, 2)
+	s.Require().NoError(err, "error should be nil")
+}
 
-	ctx, cancel := context.WithCancel(context.Background())
-	defer cancel()
+func (s *TasksServiceTestSuite) TestTerminate() {
+	// Test create ingestion task -> get status -> terminate sequence.
+	taskID, err := triggerIngestionTask(s.client, "test-terminate-task-datasource", testObjects)
+	s.Require().NoError(err, "error should be nil")
 
-	// Set up druid service and client.
-	d, err := NewClient("http://localhost:8888")
-	require.NoError(t, err, "error should be nil")
+	err = awaitTaskStatus(s.client, taskID, "RUNNING", 180*time.Second, 200*time.Millisecond)
+	s.Require().NoError(err, "error should be nil")
 
-	// Waiting for druid services to start.
-	err = compose.
-		WaitForService("coordinator", wait.NewHTTPStrategy(processInformationPathPrefix).WithPort("8081/tcp").WithStartupTimeout(180*time.Second)).
-		WaitForService("router", wait.NewHTTPStrategy(processInformationPathPrefix).WithPort("8888/tcp").WithStartupTimeout(180*time.Second)).
-		WaitForService("broker", wait.NewHTTPStrategy(processInformationPathPrefix).WithPort("8082/tcp").WithStartupTimeout(180*time.Second)).
-		WaitForService("middlemanager", wait.NewHTTPStrategy(processInformationPathPrefix).WithPort("8091/tcp").WithStartupTimeout(180*time.Second)).
-		Up(ctx, tc.Wait(true))
-	require.NoError(t, err, "druid services should be up with no error")
+	shutdownTaskID, err := s.client.Tasks().Shutdown(taskID)
+	s.Require().NoError(err, "error should be nil")
+	s.Require().Equal(shutdownTaskID, taskID)
```

Comment:
Should we also test that the task is terminated?

Reply:
not sure if that is necessary, though I've added it